liquibase-core/src/main/java/liquibase/util/TableOutput.java

```diff
@@ -201,6 +208,47 @@ public static void formatOutput(String[][] table, List<Integer> maxWidths, boole
         }
     }
 
+    /**
+     * Compute the size of the largest string of each column of the provided table
+     * @param rows the provided table to compute widths from
+     * @return an empty immutable list if the provided table is empty
+     * @throws RuntimeException if rows is null or the column count is not the same for every row
+     */
+    public static List<Integer> computeMaxWidths(List<List<String>> rows) {
+        return computeMaxWidths(rows.stream().map(row -> row.toArray(new String[0])).toArray(String[][]::new));
+    }
+
+    /**
+     * Compute the size of the largest string of each column of the provided table
+     * @param rows the provided table to compute widths from
+     * @return an empty immutable list if the provided table is empty
+     * @throws RuntimeException if rows is null or the column count is not the same for every row
+     */
+    public static List<Integer> computeMaxWidths(String[][] rows) {
+        if (rows.length == 0) {
+            return Collections.emptyList();
+        }
+        int columnCount = rows[0].length;
+        List<Integer> widths = new ArrayList<>(Collections.nCopies(columnCount, 0));
+        for (String[] row : rows) {
+            if (row.length != columnCount) {
```

Comment:
"Table Output" test (from TableOutputTests) is failing when having a single line table as this one `[["data1","data2"]]` a new empty line is being added to `finalTableList` which is failing into this if block ending up on below exception.

Reply:
Fixed in a346e50baa0f06b118c2a6ee8aa5acce9fe76910 by adding n empty strings, instead of just 1, since the logic after that already handles the "all cells empty" case and having homogeneous rows works better with `computeMaxWidths`.